federjs/Feder.js

```diff
@@ -0,0 +1,75 @@
+import FederCore from './FederCore';
+import FederView from './FederView';
+export default class Feder {
+  constructor({
+    core = null,
+    filePath = '',
+    source = '',
+    projectParams = {},
+    domSelector,
+    viewParams = {},
+  }) {
+    this.federView = new FederView({ domSelector, viewParams });
+    this.viewParams = viewParams;
+    if (!core) {
+      this.initCorePromise = fetch(filePath)
+        .then((res) => res.arrayBuffer())
+        .then((data) => {
+          core = new FederCore({ data, source, projectParams });
+          this.core = core;
+          const indexType = core.indexType;
+          const indexMeta = core.indexMeta;
+          const getVectorById = (id) =>
+            id in core.id2vector ? core.id2vector[id] : null;
+          this.federView.initView({
+            indexType,
+            indexMeta,
+            getVectorById,
+          });
+        });
+    } else {
+      // todo
+    }
+  }
+
+  async overview() {
+    this.initCorePromise && (await this.initCorePromise);
```

Comment:
if we run initCorePromise in class constructor, why we also run it in all functions ?

Reply:
just await, not rerun.